src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -788,11 +776,11 @@ internal Binder WithPatternVariablesIfAny(ArgumentListSyntax initializerArgument
             patterns.Free();
         }
 
-        internal BoundExpression WrapWithVariablesIfAny(BoundExpression expression)
+        internal BoundExpression WrapWithVariablesIfAny(CSharpSyntaxNode node, BoundExpression expression)
         {
             return (Locals.Length == 0)
                 ? expression
-                : new BoundSequence(expression.Syntax, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
+                : new BoundSequence(node, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
```

Comment:
@AlekseyTs What I mean is, we don't expect this to generate a sequence point, right? I presume we can set a break point on a pattern match, but only for the bound expression (or stmt, depending on the construct)?